### PR TITLE
`GHA` `concurrency`

### LINKED ISSUE #5190: PRs cancel CI on push
The `cancel` step doesn't seem to be configured properly, a push to a pull request shouldn't cancel CI on `master`. For reference, here are the logs for two examples: [CI Additional](https://github.com/pydata/xarray/runs/2384577555?check_suite_focus=true), [CI](https://github.com/pydata/xarray/runs/2384577560?check_suite_focus=true).

### PR DESCRIPTION
**Pull request by aaronspring**
- [x] Closes #5190
- [ ] Tests added
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency 

`concurrency` instead of `cancel-duplicate-runs.yaml`

### CODE REVIEW

**Comment by max-sixty** [comment]
Lgtm! Thanks @aaronspring

**Review by keewis — APPROVED** [review]
yep, this is great! Thanks, @aaronspring

**Review by andersy005 — APPROVED** [review]
This looks great, @aaronspring! It's my understanding that the `concurrency` ensures that there are no duplicate jobs for the same `ref`. I was wondering if we still need the `skip-duplicate-jobs` as part of the CI?

**Comment by aaronspring** [comment]
In my understanding we can get rid of fkirc/skip-duplicate-actions and cancel-duplicate-runs.yaml

File: .github/workflows/ci-additional.yaml [hunk 1]
```diff
@@ -8,27 +8,14 @@ on:
       - "*"
   workflow_dispatch: # allows you to trigger manually
 
+concurrency:
```
Comment (aaronspring) [inline .github/workflows/ci-additional.yaml:11]:
```suggestion
concurrency:
```

File: .github/workflows/upstream-dev-ci.yaml [hunk 2]
```diff
@@ -10,6 +10,10 @@ on:
     - cron: "0 0 * * *" # Daily “At 00:00” UTC
   workflow_dispatch: # allows you to trigger the workflow run manually
 
+concurrency:
```
Comment (aaronspring) [inline .github/workflows/upstream-dev-ci.yaml:13]:
```suggestion
concurrency:
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -8,27 +8,13 @@ on:
       - "*"
   workflow_dispatch: # allows you to trigger manually
 
-jobs:
-  skip-duplicate-jobs:
-    runs-on: ubuntu-latest
-    if: |
-      github.repository == 'pydata/xarray'
-      && (github.event_name == 'push' || github.event_name == 'pull_request')
```
Comment (andersy005) [inline .github/workflows/ci.yaml:16]: @aaronspring, we may want to keep this condition so as to avoid running the CI on forks

**Comment by max-sixty** [comment]
Great — merging, thanks @aaronspring !

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (aaronspring) [reply .github/workflows/ci.yaml:16]: Shouldn’t we keep this? @max-sixty @andersy005

Comment (keewis) [reply .github/workflows/ci.yaml:16]: we should! @aaronspring, can you open a new PR for that?

Comment (max-sixty) [reply .github/workflows/ci.yaml:16]: Sorry, I merged too early. @aaronspring if you can open a PR then great, otherwise I can as penance

Comment (aaronspring) [reply .github/workflows/ci.yaml:16]: I put the condition in `detect-ci-trigger` in #6223

**Pull request merged**